deployment/docker-compose.yml

```
@@ -0,0 +1,182 @@
+      - "traefik.http.routers.airflow.rule=Host(`airflow.${PUBLIC_HOSTNAME}`)"
+      - "traefik.http.routers.airflow.entrypoints=websecure"
+      - "traefik.http.routers.airflow.tls.certresolver=main"
+    depends_on:
+      <<: *airflow-common-depends-on
+      airflow-init:
+        condition: service_completed_successfully
+
+  airflow-scheduler:
+    <<: *airflow-common
+    command: scheduler
+    restart: on-failure
+    healthcheck:
+      test:
+        [
+          "CMD-SHELL",
+          'airflow jobs check --job-type SchedulerJob --hostname "$${HOSTNAME}"'
+        ]
+      interval: 10s
+      timeout: 10s
+      retries: 5
+    depends_on:
+      <<: *airflow-common-depends-on
+      airflow-init:
+        condition: service_completed_successfully
+
+  dbt-init:
+    image: ghcr.io/dbt-labs/dbt-postgres:1.5.1
+    command: deps
+    environment:
+      DBT_PROFILES_DIR: /usr/app
+    volumes:
+      - ../pipeline/dbt:/usr/app
+
+  datawarehouse:
+    image: ghcr.io/betagouv/data-inclusion-datawarehouse:${STACK_VERSION}
+    restart: on-failure
+    healthcheck:
+      test: [ "CMD", "pg_isready", "-U", "${DATAWAREHOUSE_DI_USERNAME}"]
+      interval: 5s
+      retries: 5
+    ports:
+      - 5432:5432
+    environment:
+      - POSTGRES_DB=${DATAWAREHOUSE_DI_DATABASE}
+      - POSTGRES_USER=${DATAWAREHOUSE_DI_USERNAME}
+      - POSTGRES_PASSWORD=${DATAWAREHOUSE_DI_PASSWORD}
+    volumes:
+      - datawarehouse-data:/var/lib/postgresql/data
+
+  api:
+    image: ghcr.io/betagouv/data-inclusion-api:${STACK_VERSION}
+    depends_on:
+      - datawarehouse
+    restart: always
+    ports:
+      - 8000:8000
+    labels:
+      - "traefik.enable=true"
+      - "traefik.http.routers.api.rule=Host(`api.${PUBLIC_HOSTNAME}`)"
+      - "traefik.http.routers.api.entrypoints=websecure"
+      - "traefik.http.routers.api.tls.certresolver=main"
+    environment:
+      - ENV=prod
+      - DEBUG=False
+      - DATABASE_URL=${AIRFLOW_CONN_PG}  # TODO: create dedicated RO creds for api
+      - SECRET_KEY=${API_SECRET_KEY}
+      - ROOT_PATH=/api
+      - TOKEN_ENABLED=${API_TOKEN_ENABLED}
+
+  reverse-proxy:
+    image: traefik:v2.10
+    ports:
+      - 80:80
+      - 443:443
+      - 8081:8080
+    environment:
+      # - TRAEFIK_API_INSECURE=true
+      # - TRAEFIK_LOG_LEVEL=debug
+      - TRAEFIK_PROVIDERS_DOCKER=true
+      - TRAEFIK_PROVIDERS_DOCKER_EXPOSEDBYDEFAULT=false
+      - TRAEFIK_ENTRYPOINTS_WEB_ADDRESS=:80
+      - TRAEFIK_ENTRYPOINTS_WEB_HTTP_REDIRECTIONS_ENTRYPOINT_TO=websecure
+      - TRAEFIK_ENTRYPOINTS_WEBSECURE_ADDRESS=:443
+      - TRAEFIK_CERTIFICATESRESOLVERS_MAIN_ACME_TLSCHALLENGE=true
+      - TRAEFIK_CERTIFICATESRESOLVERS_MAIN_ACME_EMAIL=tech@data.inclusion.beta.gouv.fr
+      - TRAEFIK_CERTIFICATESRESOLVERS_MAIN_ACME_STORAGE=/letsencrypt/acme.json
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock:ro
+      - ./letsencrypt:/letsencrypt
+
+volumes:
+  airflow-logs:
+  datawarehouse-data:
+
```

Je suggère d'utiliser un volume persistent pour le `acme.json` histoire de ne pas tomber sur une rate limit Let's Encrypt si on renouvelle le certificat à chaque déploiement :

```suggestion
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - letsencrypt-data:/letsencrypt

volumes:
  airflow-logs:
  datawarehouse-data:
  letsencrypt-data:
```